language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CodeActionTest.java

```diff
@@ -213,7 +256,15 @@ public void testCodeActionWithTestGen(String config, Path source) throws IOExcep
                 {"undefinedFunctionCodeAction2.json", "createUndefinedFunction2.bal"},
                 {"variableAssignmentRequiredCodeAction.json", "createVariable.bal"},
                 {"ignoreReturnValueCodeAction.json", "createVariable.bal"},
-                {"packagePull.json", "packagePull.bal"}
+                {"packagePull.json", "packagePull.bal"},
+        };
+    }
+
+    @DataProvider(name = "codeaction-quickfixes-data-provider")
+    public Object[][] codeActionQuickFixesDataProvider() {
+        log.info("Test textDocument/codeAction QuickFixes");
+        return new Object[][]{
+                {"markUntaintedCodeAction.json", "taintedVariable.bal"}
```

Comment:
Let's add a couple of more test cases to address variable definition related options as well

Reply:
This sample covers all cases defined in [taint_checking.bal](https://github.com/ballerina-platform/ballerina-lang/blob/d308040ef21b7968c04e140f7ddf922369178a5c/examples/taint-checking/taint_checking.bal)